### PR TITLE
samples: net: Add civetweb HTTP sample

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**
This commit adds civetweb as a west module and a sample that uses it.

Signed-off-by: Tomasz Gorochowik <tgorochowik@antmicro.com>

### CODE REVIEW

File: samples/net/sockets/civetweb/src/libc_extensions.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	return 0;
+}
+
+double difftime (time_t end, time_t beg)
+{
+	return end - beg;
+}
+
+static char* generic_strerr = "[STRERR NOT INMPLEMENTED]";
```
Comment (PiotrZierhoffer) [inline samples/net/sockets/civetweb/src/libc_extensions.c]: IMPLEMENTED

Comment (PiotrZierhoffer) [reply samples/net/sockets/civetweb/src/libc_extensions.c]: Maybe just "`ERRNO %d", err`, if we can do it easily? Then it would at least be helpful

Comment (PiotrZierhoffer) [reply samples/net/sockets/civetweb/src/libc_extensions.c]: can be preallocated even

File: samples/net/sockets/civetweb/src/libc_extensions.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+double atof(const char* str)
+{
+	/* XXX good enough for civetweb uses */
+	return (double)atoi(str);
+}
+
+long long int strtoll(const char* str, char** endptr, int base)
+{
+	FN_MISSING();
```
Comment (PiotrZierhoffer) [inline samples/net/sockets/civetweb/src/libc_extensions.c]: will strtoul be enough for civetweb? It is implemented

File: samples/net/sockets/civetweb/src/main.c [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+void send_ok(struct mg_connection *conn)
+{
+	mg_printf(conn,
+		  "HTTP/1.1 200 OK\r\n"
+		  "Content-Type: text/html\r\n"
+		  "Connection: close\r\n\r\n");
+}
+
+int hello_world_handler(struct mg_connection *conn, void *cbdata)
```
Comment (PiotrZierhoffer) [inline samples/net/sockets/civetweb/src/main.c:36]: maybe we could group it with other handlers

File: samples/net/sockets/civetweb/src/libc_extensions.c (hunk 2 above)
Comment (tgorochowik) [reply samples/net/sockets/civetweb/src/libc_extensions.c]: It should, right!

**Pull request closed without merging**